File: java/src/test/java/com/kcl/GetVersionTest.java
package com.kcl;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.GetVersionArgs;
import com.kcl.api.Spec.GetVersionResult;

public class GetVersionTest {
    @Test
    public void testGetVersion() throws Exception {
        API api = new API();
        GetVersionArgs version_args = GetVersionArgs.newBuilder().build();
        GetVersionResult result = api.getVersion(version_args);
        String versionInfo = result.getVersionInfo();
        Assert.assertTrue(versionInfo, versionInfo.contains("Version"));
        Assert.assertTrue(versionInfo, versionInfo.contains("GitCommit"));
    }
}

File: kotlin/src/test/kotlin/com/kcl/GetVersionTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.getVersionArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class GetVersionTest {
    @Test
    fun testGetVersion() {
        val args = getVersionArgs {}
        val api = API()
        val result = api.getVersion(args)
        assertEquals(result.versionInfo.contains("Version"), true)
        assertEquals(result.versionInfo.contains("GitCommit"), true)
    }
}
